spec/02-integration/04-admin_api/21-client_certificate.lua

```diff
@@ -0,0 +1,226 @@
+jlxKAveN5WizYAEFxQ9brPsEuwXzc0paCyHNqYtQm1WgzFbDlCj+3MMTfEb1xCol
+hG5JLaRtW8q6njH7gIhjwcSuz3Hcfri0qqjSIvSn1CEUCKNUOO1B2ip8Dm86ZZSf
+1jWDxOyIfJctmk+YHYxzispLnbw21iNW726tVQhA5M0LXAkIFUO+lS/Skk6Q/hct
+pZQhmTkNAgMBAAECggEAQkWKa9uKalYNHNC3dAMlYXXSHk9Ek6YL8Ykd5rY15Lnf
+bOHPtva4XnjdGkoQ9BBf333JfgFXRbt8bL+NFlff/+KVPT1WHYbmdere9bYAegwR
+0j8V9+j6FnMMQR03+3SMeVk2Ic4ZNu/0m6263gzPrwEWwAJc8wPNuYnWqTlSfI8V
+hG/hB7TB/Yz3HbWg+rXE84RsT280MG0gzkK9U1dQVGYuPAfiqZt9ZwQ9HjbVjVLu
+20NqBIzFE4/IV/wXVNHrSUdE8GLuYxlO1uKg6h71FDBsEu3lHyx4S0xhADbZIREr
+ZwY1+PCAL3nuUUc2KuEU1Bc2tjBWgasY2kLioxDzRQKBgQDTA+4TjUzuKbbeZBn6
+m7g+lyK9I/lf+hge++erYzF64WKgbD8cuZmS4c3N5b3iH0pUdc6s6c1eERB+DzqG
+RraB9rD0bqFYWinIEqPvDmzxAnkgNHlar9NGpi4Ow5GHvLCY5POI4/fd9dBOPeKG
+IYJ78rXhYinBkIyaeNv7iVQwRwKBgQD2gn560wekurtzKz/2kOisvwUI5UDR4n5Q
+aypwBO/XoVVdlmKkzZuEq4EPDnxP0WpVAkpVRwYgv0JBL0TblvZEf75dftuvH0g4
+MerD6pWQV9NjMhDfGN9EnomhF4qN2PxfH6Ae8voa7XUgBGrANTVzQ28r+kIxgBhB
+UmcdF7SqCwKBgHxQF2S7XLP+LTZtAIUMUgnFnfiRhetN+gHe44erg03iE+lwe+nf
+SaMnlcgT7b6WMoL5eE2T26HsKawqKo9Kf2CnRW5q8nCLhYvRt3QxKmr9yv/3Cfss
+FCrN4dpeypCNz/fzAaoGq1K+6v39JUNh4G5G+XGTFAhtsPeb9fUE3W0nAoGBAPSe
+JfioNSuIib8Nqhl5TwstO0fKh3o/ce1HiwNCQucNl/KY6yZzSeJrdlSEoWNrgkL9
+VNou8qvIX2zkBUaLFbJ/uV56JnIvbrElB7yGqwhjvsbuxcM9oUEsuZJ8E+gS7M1Q
+Ssqrfvj9rn6pI22831+UCVvwqYMJhAg+R0idX5BlAoGAY6lp43yoEBg7+JEvzrYq
+cTbqrL0lo4W9JsDai65lJ/f6QTP1I53vHiUBUamnbuBD/dbn2CWR0vbkM1gbkM6M
+/rEJqU5K54qS+IDxd27YsTy9SdSTE6UakYn/bweVTed2qmnY1SQZUt1KWThwK4xx
+XeHHwxnzI/T07RdxJufNKbQ=
+-----END PRIVATE KEY-----
+]]
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("Cert caching update test ->", function ()
+    local admin_client, cert_id
+
+    lazy_setup(function()
+      local bp, db = helpers.get_db_utils(strategy)
+      local fixtures = {
+        dns_mock = helpers.dns_mock.new(),
+        http_mock = {
+          ssl_response = [[
+            server {
+              server_name test.com;
+              listen 443 ssl;
+
+              ssl_certificate     /etc/aaa.com.cert.pem;
+              ssl_certificate_key /etc/aaa.com.key.pem;
+
+              ssl_verify_client on;
+              ssl_client_certificate /etc/rootCA.pem;
+              ssl_verify_depth 2;
+
+              body_filter_by_lua_block {
+                  ngx.arg[1] = ngx.var.ssl_client_escaped_cert
+              }
+            }
+          ]]
+        }
+      }
+
+      fixtures.dns_mock:A {
+        name = "test.com",
+        address = "127.0.0.1",
+      }
+
+      -- create cert
+      local certObj = bp.certificates:insert({
+        cert = cert,
+        key = key,
+      })
+      assert.not_nil(certObj)
+      cert_id = certObj.id
+
+      -- create service
+      local srvObj = bp.services:insert({
+        name = "nginx",
+        host = "test.com",
+        protocol = "https",
+        port = 443,
+        client_certificate = { id = cert_id },
+      })
+      assert.not_nil(srvObj)
+
+      -- create route
+      local routeObj = bp.routes:insert({
+        name = "ngxin",
```

Comment:
typo

Reply:
Should be snake_case?

Reply:
tiny typo
```suggestion
        name = "nginx",
```

Reply:
should be `nginx`

Reply:
Also, snake_case is encouraged in kong repo.